projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala

```diff
@@ -102,10 +112,17 @@ class ProjectionRegistryActor extends Actor with ActorLogging {
       // keep track
       val workerKey: WorkerKey = coordinates.asKey
       updateNameIndex(workerKey, coordinates)
-      actorIndex = actorIndex.updated(workerKey, sender())
+      localActorIndex = localActorIndex.updated(workerKey, sender())
       reversedActorIndex = reversedActorIndex.updated(sender, workerKey)
-      // when worker registers, we must reply with the requested status (if it's been set already, or DefaultInitialStatus if not).
-      val initialStatus = requestedStatusLocalCopy.getOrElse(workerKey, DefaultRequestedStatus)
+      // when worker registers, we must reply with the requested status (if it's been set already). If no
+      // specific request for the worker is found, then fallback to projection-level requests. Finally, fallback to
+      // DefaultInitialStatus.
+      val initialStatus =
+        workerRequestedStatusLocalCopy.getOrElse(
+          workerKey,
+          projectionRequestedStatusLocalCopy.getOrElse(coordinates.projectionName, DefaultRequestedStatus)
+        )
```

Comment:
This is the main improvement: check worker intent, otherwise fallback to projection intent, otherwise fallback to default initial status.